Update system_test_python_versions to fix "ResourceWarning: unclosed event loop" errors from grpc

1. Tests failing after release of grpcio library: https://pypi.org/project/grpcio/1.62.0/#history
2. Occuring because we use Python 3.10 but grpcio expects Python3.12


Error logs:

```
             warnings.warn(pytest.PytestUnraisableExceptionWarning(msg))
E                   pytest.PytestUnraisableExceptionWarning: Exception ignored in: <socket.socket fd=-1, family=1, type=1, proto=0>
E                   
E                   Traceback (most recent call last):
E                     File "src/python/grpcio/grpc/_cython/_cygrpc/aio/common.pyx.pxi", line 184, in grpc._cython.cygrpc.get_working_loop
E                   RuntimeError: no running event loop
E                   
E                   During handling of the above exception, another exception occurred:
E                   
E                   Traceback (most recent call last):
E                     File "src/python/grpcio/grpc/_cython/_cygrpc/aio/common.pyx.pxi", line 190, in grpc._cython.cygrpc.get_working_loop
E                     File "/opt/hostedtoolcache/Python/3.12.2/x64/lib/python3.12/asyncio/events.py", line 697, in get_event_loop
E                       warnings.warn('There is no current event loop',
E                   DeprecationWarning: There is no current event loop
E                   
E                   During handling of the above exception, another exception occurred:
E                   
E                   Traceback (most recent call last):
E                     File "<frozen _collections_abc>", line 804, in get
E                   ResourceWarning: unclosed <socket.socket fd=26, family=1, type=1, proto=0>

```

3. Sample [failure](https://github.com/googleapis/python-pubsub/actions/runs/8011050674/job/21883416693?pr=1087)

4. Related issue created in grpc repo: https://github.com/grpc/grpc/issues/35974

--- a/CONTRIBUTING.rst
+++ b/CONTRIBUTING.rst
@@ -143,12 +143,12 @@ Running System Tests
    $ nox -s system
 
    # Run a single system test
-   $ nox -s system-3.10 -- -k <name of test>
+   $ nox -s system-3.12 -- -k <name of test>
 
 
   .. note::
 
-      System tests are only configured to run under Python 3.10.
+      System tests are only configured to run under Python 3.12.
       For expediency, we do not run them in older versions of Python 3.
 
   This alone will not run the tests. You'll need to change some local

--- a/noxfile.py
+++ b/noxfile.py
@@ -50,7 +50,7 @@ UNIT_TEST_DEPENDENCIES: List[str] = []
 UNIT_TEST_EXTRAS: List[str] = []
 UNIT_TEST_EXTRAS_BY_PYTHON: Dict[str, List[str]] = {}
 
-SYSTEM_TEST_PYTHON_VERSIONS: List[str] = ["3.10"]
+SYSTEM_TEST_PYTHON_VERSIONS: List[str] = ["3.12"]
 SYSTEM_TEST_STANDARD_DEPENDENCIES: List[str] = [
     "mock",
     "pytest",

--- a/owlbot.py
+++ b/owlbot.py
@@ -337,7 +337,7 @@ templated_files = gcp.CommonTemplates().py_library(
     cov_level=100,
     versions=gcp.common.detect_versions(path="./google", default_first=True),
     unit_test_python_versions=["3.7", "3.8", "3.9", "3.10", "3.11", "3.12"],
-    system_test_python_versions=["3.10"],
+    system_test_python_versions=["3.12"],
     system_test_external_dependencies=["psutil","flaky"],
 )
 s.move(templated_files, excludes=[".coveragerc", ".github/release-please.yml", "README.rst", "docs/index.rst"])

--- a/pytest.ini
+++ b/pytest.ini
@@ -12,3 +12,5 @@ filterwarnings =
     ignore:The return_immediately flag is deprecated and should be set to False.:DeprecationWarning
     # Remove warning once https://github.com/googleapis/gapic-generator-python/issues/1939 is fixed
     ignore:get_mtls_endpoint_and_cert_source is deprecated.:DeprecationWarning
+    # Remove warning once https://github.com/grpc/grpc/issues/35974 is fixed
+    ignore:unclosed:ResourceWarning

